CMakeLists.txt

```diff
@@ -251,7 +251,7 @@ if(MSVC)
   add_definitions(-DWIN32)
 else()
   add_definitions(-Wall -Wextra -pedantic -Wno-unused-parameter
-    -Wstrict-prototypes -std=gnu99)
+    -Wno-tautological-compare -Wstrict-prototypes -std=gnu99)
```

Comment:
It would be better if we could limit this to the files we know might trigger this warning, or fix the macros to not trigger it.

Reply:
This is a false positive in the version of clang-7 that's currently in Xenial.  Since we've switched to clang-4.0, I removed this.  When we switch back to a newer clang, this won't be an issue.